plugins/copr.py

```diff
@@ -473,34 +477,40 @@ def _download_repo(self, project_name, repo_filename):
+            response = urlopen(self.copr_url + api_path)
             if os.path.exists(repo_filename):
                 os.remove(repo_filename)
-            if '404' in str(e):
-                if PY3:
-                    import urllib.request
-                    try:
-                        res = urllib.request.urlopen(self.copr_url + "/coprs/" + project_name)
-                        status_code = res.getcode()
-                    except urllib.error.HTTPError as e:
-                        status_code = e.getcode()
-                else:
-                    import urllib
-                    res = urllib.urlopen(self.copr_url + "/coprs/" + project_name)
-                    status_code = res.getcode()
-                if str(status_code) != '404':
-                    raise dnf.exceptions.Error(_("This repository does not have"
-                                                 " any builds yet so you cannot enable it now."))
-                else:
-                    raise dnf.exceptions.Error(_("Such repository does not exist."))
-            raise
-
-        for line in f:
-            if re.match(r"\[copr:", line):
-                repo_filename = os.path.join(self.base.conf.get_reposdir,
-                                             "_" + line[1:-2] + ".repo")
-            break
+        except HTTPError as e:
+            if e.code != 404:
+                error_msg = _("Request to {0} failed: {1} - {2}").format(self.copr_url + api_path, e.code, str(e))
+                raise dnf.exceptions.Error(error_msg)
+            error_msg = _("It wasn't possible to enable this project.\n")
+            error_data = e.headers.get("Copr-Error-Data")
+            if error_data:
+                error_data_decoded = base64.b64decode(error_data).decode('utf-8')
+                error_data_decoded = json.loads(error_data_decoded)
+                error_msg += _("Repository '{0}' does not exist in project '{1}'.").format(
+                    '-'.join(self.chroot_parts), project_name)
+                if error_data_decoded.get("available chroots"):
+                    error_msg += _("\nAvailable repositories: ") + ', '.join(
+                        "'{}'".format(x) for x in error_data_decoded["available chroots"])
+                    error_msg += _("\n\nIf you want to enable a non-default repository, use the following command:\n"
+                                   "  'dnf copr enable {0} <repository>'\n"
+                                   "But note that the installed repo file will likely need a manual "
+                                   "modification.").format(project_name)
+                raise dnf.exceptions.Error(error_msg)
+            else:
+                error_msg += _("Project {0} does not exist.").format(project_name)
+                raise dnf.exceptions.Error(error_msg)
+        except URLError as e:
+            error_msg = _("Failed to connect to {0}: {1}").format(self.copr_url + api_path, e.reason.strerror)
```

Comment:
Ad the `if` comment above, ... I think we can calmly use `error_msg +=` here, right?

Reply:
I wouldn't say that. If we need to handle the URLErrort exception, then line 506 is the first line where the `error_msg` variable is defined.

Reply:
Indeed, I had to look at the whole file (not just diff) to see this.  Then LGTM.